### DISCUSSION TITLE
Writing custom endpoints in Typescript

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#21340] [Feature Requests]
### Describe the Bug

Hi, after more than a couple of years we're really (still) struggling with this issue I opened on October 2021:
[Writing custom endpoints in Typescript](https://github.com/directus/directus/discussions/9233)

Our product is based around an installation of Directus that we heavily expanded with around 100k lines of code we used to program some 200 custom endpoints. Our daily work is to expand on them and fix bugs on this pretty large codebase. 

Our project used to be organized like this:

```
/src
  /extensions
    /endpoints
      /custom
        /namespace1
          /utils
          /other-directory
          /additional-code
          index.js
        /namespace2
          /utils
          /other-directory
          /additional-code
          index.js
        /namespace3
          /utils
          /other-directory
          /additional-code
          index.js
        ...   
        /common-utils
        /libs
        /constants
        ...
```

Directus starts up, discovers the custom endpoints and mounts them. We've worked for almost three years in this way and we rarely had any problem.

When we switched to Directus 10 we converted our codebase to Typescript.

In our ideal world Directus would "just work" out of the box. Our frontend, for instance, is a React app build with [Create react app](https://create-react-app.dev/) and Typescript and with minimal configurations we're just one `npm run start` away from having it compiling / watching / serving / creating the sourcemaps for us.

With Directus we had to painstakingly set up a [esbuild](https://esbuild.github.io/) chain with a watcher / builder that scans our `extensions` directory and builds it to a `dist` directory that we configured Directus to use. This setup is slow, it eats up a lot of memory because it forces us to have a couple of processes running (the builder and Directus) and they don't work well together because when changes happen on any file, both Directus and the compiler detect a changed file but often the Directus process re-starts _before_ the compiler has finished compiling, effectively losing the changes and forcing us to monitor a couple of consoles whenever we save a file to manually force the restart.

Also, both [Visual studio code](https://code.visualstudio.com/) and [WebStorm](https://www.jetbrains.com/webstorm/) (the two IDEs we use in our team) often -randomly - don't allow us to set breakpoint with the new setup anymore.

I'm writing this issue on behalf of my team because "writing custom extensions with Typescript" seems to be a basic functionality and we heavily relay upon it, but we're having trouble setting it up to work seamlessly with Directus and we need help. What's the suggested way to do so? What are other people doing? Lately, developing Directus has become a pain to us.

Thanks for any support you'll be able to provide.

### To Reproduce

Try to write custom endpoints in Typescript.

### Directus Version

10.9.0

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by br41nslug** [comment]
Converting this to a discussion as it is more of a question / observation / support request than a singular / actionable bug.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
